### DISCUSSION TITLE
How to parse escaped nginx log line with escaped double quotes using VRL

### DISCUSSION THREAD

**Discussion by prabhatsharma** [vectordotdev/vector#17885] [Q&A] [answered]
This is my source data:

```json
{
  "_timestamp": 1688641704414631,
  "kubernetes_annotations_kubernetes_io_psp": "eks.privileged",
  "kubernetes_container_hash": "registry.k8s.io/ingress-nginx/controller@sha256:4ba73c697770664c1e00e9f968de14e08f606ff961c76e5d7033a4a9c593c629",
  "kubernetes_container_image": "registry.k8s.io/ingress-nginx/controller@sha256:4ba73c697770664c1e00e9f968de14e08f606ff961c76e5d7033a4a9c593c629",
  "kubernetes_container_name": "controller",
  "kubernetes_labels_app_kubernetes_io_component": "controller",
  "kubernetes_labels_app_kubernetes_io_instance": "ingress-nginx",
  "kubernetes_labels_app_kubernetes_io_name": "ingress-nginx",
  "kubernetes_labels_pod_template_hash": "7d97444d5",
  "kubernetes_namespace_name": "ingress-nginx",
  "kubernetes_pod_name": "ingress-nginx-controller-7d97444d5-l9rfd",
  "log": "10.2.81.141 - alice [06/Jul/2023:11:08:24 +0000] \"POST /api/path1/prometheus/api/v1/write HTTP/1.1\" 200 0 \"-\" \"Prometheus/2.15.2+ds\" 468 0.005 [zinc-cp1-zinc-cp-4082] [] 9.9.176.197:4082 0 0.005 200 0ca2f386ce3b27951f36fb1f54af41f5\n",
  "stream": "stdout"
}
```

log field is standard nginx log line.

How do I parse this using VRL with parse_nginx_log?

**Comment by jszwedko** [comment]
You should be able to do something like `parse_nginx_log(.log, format: "combined")`.

**Comment by prabhatsharma** [comment]
I have tried parse_nginx_log(.log, format: "combined") earlier. It [fails](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLmxvZzEgPSBwYXJzZV9uZ2lueF9sb2chKC5sb2csIGZvcm1hdDogXCJjb21iaW5lZFwiKSIsImV2ZW50Ijp7Il90aW1lc3RhbXAiOjE2ODg2NDE3MDQ0MTQ2MzEsImt1YmVybmV0ZXNfYW5ub3RhdGlvbnNfa3ViZXJuZXRlc19pb19wc3AiOiJla3MucHJpdmlsZWdlZCIsImt1YmVybmV0ZXNfY29udGFpbmVyX2hhc2giOiJyZWdpc3RyeS5rOHMuaW8vaW5ncmVzcy1uZ2lueC9jb250cm9sbGVyQHNoYTI1Njo0YmE3M2M2OTc3NzA2NjRjMWUwMGU5Zjk2OGRlMTRlMDhmNjA2ZmY5NjFjNzZlNWQ3MDMzYTRhOWM1OTNjNjI5Iiwia3ViZXJuZXRlc19jb250YWluZXJfaW1hZ2UiOiJyZWdpc3RyeS5rOHMuaW8vaW5ncmVzcy1uZ2lueC9jb250cm9sbGVyQHNoYTI1Njo0YmE3M2M2OTc3NzA2NjRjMWUwMGU5Zjk2OGRlMTRlMDhmNjA2ZmY5NjFjNzZlNWQ3MDMzYTRhOWM1OTNjNjI5Iiwia3ViZXJuZXRlc19jb250YWluZXJfbmFtZSI6ImNvbnRyb2xsZXIiLCJrdWJlcm5ldGVzX2xhYmVsc19hcHBfa3ViZXJuZXRlc19pb19jb21wb25lbnQiOiJjb250cm9sbGVyIiwia3ViZXJuZXRlc19sYWJlbHNfYXBwX2t1YmVybmV0ZXNfaW9faW5zdGFuY2UiOiJpbmdyZXNzLW5naW54Iiwia3ViZXJuZXRlc19sYWJlbHNfYXBwX2t1YmVybmV0ZXNfaW9fbmFtZSI6ImluZ3Jlc3MtbmdpbngiLCJrdWJlcm5ldGVzX2xhYmVsc19wb2RfdGVtcGxhdGVfaGFzaCI6IjdkOTc0NDRkNSIsImt1YmVybmV0ZXNfbmFtZXNwYWNlX25hbWUiOiJpbmdyZXNzLW5naW54Iiwia3ViZXJuZXRlc19wb2RfbmFtZSI6ImluZ3Jlc3MtbmdpbngtY29udHJvbGxlci03ZDk3NDQ0ZDUtbDlyZmQiLCJsb2ciOiIxMC4yLjgxLjE0MSAtIGFsaWNlIFswNi9KdWwvMjAyMzoxMTowODoyNCArMDAwMF0gXCJQT1NUIC9hcGkvcGF0aDEvcHJvbWV0aGV1cy9hcGkvdjEvd3JpdGUgSFRUUC8xLjFcIiAyMDAgMCBcIi1cIiBcIlByb21ldGhldXMvMi4xNS4yK2RzXCIgNDY4IDAuMDA1IFt6aW5jLWNwMS16aW5jLWNwLTQwODJdIFtdIDkuOS4xNzYuMTk3OjQwODIgMCAwLjAwNSAyMDAgMGNhMmYzODZjZTNiMjc5NTFmMzZmYjFmNTRhZjQxZjVcbiIsInN0cmVhbSI6InN0ZG91dCJ9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D)

Probably because of the escaped double quotes

**Comment by dsmith3197** [comment, marked as answer]
I was able to successfully parse the example log line in the docs with escaped double quotes. For your case, you'll want to use the `ingress_upstreaminfo` format ([example](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLmxvZzEgPSBwYXJzZV9uZ2lueF9sb2chKC5sb2csIGZvcm1hdDogXCJpbmdyZXNzX3Vwc3RyZWFtaW5mb1wiKSIsImV2ZW50Ijp7Il90aW1lc3RhbXAiOjE2ODg2NDE3MDQ0MTQ2MzEsImt1YmVybmV0ZXNfYW5ub3RhdGlvbnNfa3ViZXJuZXRlc19pb19wc3AiOiJla3MucHJpdmlsZWdlZCIsImt1YmVybmV0ZXNfY29udGFpbmVyX2hhc2giOiJyZWdpc3RyeS5rOHMuaW8vaW5ncmVzcy1uZ2lueC9jb250cm9sbGVyQHNoYTI1Njo0YmE3M2M2OTc3NzA2NjRjMWUwMGU5Zjk2OGRlMTRlMDhmNjA2ZmY5NjFjNzZlNWQ3MDMzYTRhOWM1OTNjNjI5Iiwia3ViZXJuZXRlc19jb250YWluZXJfaW1hZ2UiOiJyZWdpc3RyeS5rOHMuaW8vaW5ncmVzcy1uZ2lueC9jb250cm9sbGVyQHNoYTI1Njo0YmE3M2M2OTc3NzA2NjRjMWUwMGU5Zjk2OGRlMTRlMDhmNjA2ZmY5NjFjNzZlNWQ3MDMzYTRhOWM1OTNjNjI5Iiwia3ViZXJuZXRlc19jb250YWluZXJfbmFtZSI6ImNvbnRyb2xsZXIiLCJrdWJlcm5ldGVzX2xhYmVsc19hcHBfa3ViZXJuZXRlc19pb19jb21wb25lbnQiOiJjb250cm9sbGVyIiwia3ViZXJuZXRlc19sYWJlbHNfYXBwX2t1YmVybmV0ZXNfaW9faW5zdGFuY2UiOiJpbmdyZXNzLW5naW54Iiwia3ViZXJuZXRlc19sYWJlbHNfYXBwX2t1YmVybmV0ZXNfaW9fbmFtZSI6ImluZ3Jlc3MtbmdpbngiLCJrdWJlcm5ldGVzX2xhYmVsc19wb2RfdGVtcGxhdGVfaGFzaCI6IjdkOTc0NDRkNSIsImt1YmVybmV0ZXNfbmFtZXNwYWNlX25hbWUiOiJpbmdyZXNzLW5naW54Iiwia3ViZXJuZXRlc19wb2RfbmFtZSI6ImluZ3Jlc3MtbmdpbngtY29udHJvbGxlci03ZDk3NDQ0ZDUtbDlyZmQiLCJsb2ciOiIxMC4yLjgxLjE0MSAtIGFsaWNlIFswNi9KdWwvMjAyMzoxMTowODoyNCArMDAwMF0gXCJQT1NUIC9hcGkvcGF0aDEvcHJvbWV0aGV1cy9hcGkvdjEvd3JpdGUgSFRUUC8xLjFcIiAyMDAgMCBcIi1cIiBcIlByb21ldGhldXMvMi4xNS4yK2RzXCIgNDY4IDAuMDA1IFt6aW5jLWNwMS16aW5jLWNwLTQwODJdIFtdIDkuOS4xNzYuMTk3OjQwODIgMCAwLjAwNSAyMDAgMGNhMmYzODZjZTNiMjc5NTFmMzZmYjFmNTRhZjQxZjVcbiIsInN0cmVhbSI6InN0ZG91dCJ9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D)).

**Reply by dsmith3197** [reply]
I noticed that `ingress_upstreaminfo` is not in the public docs. Opened https://github.com/vectordotdev/vector/issues/17893 to fix this.
